app/src/main/java/com/example/una/LoginActivity.java

```diff
@@ -0,0 +1,109 @@
+package com.example.una;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.ErrorCodes;
+import com.firebase.ui.auth.IdpResponse;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class LoginActivity extends AppCompatActivity {
+
+    protected static final int RC_SIGN_IN = 123;
+
+    Context context;
+    private FirebaseAuth auth;
+    private final String TAG = this.getClass().getSimpleName();
+
+    final List<AuthUI.IdpConfig> providers = Arrays.asList(
```

Comment:
The general convention on Java projects is to have ALL_CAPS_NAMES_WITH_UNDERSCORES for `final` variables. Example: `RC_SIGN_IN` and `TAG`